core/include/vecmem/edm/host.hpp

```diff
@@ -80,6 +86,8 @@ class host<schema<VARTYPES...>, INTERFACE> {
     /// Reserve memory for the container
     VECMEM_HOST
     void reserve(size_type size);
+    /// Add a new element to the container
+    VECMEM_HOST void push_back(const object_type& element);
```

Comment:
Note that, as I discussed with @stephenswat, I was trying to make this function much more generic at first. So that instead of using `object_type`, it would use a generic proxy here. But Clang was really not happy about it. (Getting itself confused about what to do with reference variables at the end of some long templating stacks.)

So, in order to use `push_back(...)`, all variables need to be loaded onto the stack first. Even if one is copying an element from another SoA container. Not what I intended, but I **think** it should not matter much in the end.